library/formsoptions.inc.php

```diff
@@ -33,6 +33,50 @@ function checkFormIsActive ($form_name, $encounter)
 
 return $formid;
 }
+#copy this and use case num from the calendar in place of encounter.
+function FindCaseUsed ($form_name, $pid, $casenum )
+{
+   # Count the prior auths and determine the used number.
+   # Pass the PID, Auth Number, and form name.
+   # Make sure we have the correct auth.
+   $query_get_casenum = sqlquery("SELECT f.case_number, f.auth_to FROM $form_name AS f " .
+                          "WHERE pid = ? AND f.case_number = ? AND archived = '0' ", array($pid, $casenum));
+    $query_casenum_count = sqlquery("SELECT count(*) AS count FROM $form_name " .
+                          "WHERE pid = ? AND case_number = ? ", array($pid,$query_get_casenum['case_number']));
+return $query_casenum_count;
+}
+
+function FindAuthUsed ($form_name, $pid, $encounter )
+{
+   # Count the prior auths and determine the used number.
+   # Pass the PID, Auth Number, and form name.
+   # Make sure we have the correct auth.
+
+   $query_get_authnum = sqlquery("SELECT f.prior_auth_number, f.auth_to FROM $form_name AS f " .
+                          "WHERE pid = ? AND f.case_number >= ? AND archived = '0' ORDER BY f.id DESC", array($pid, date("Y-m-d")));
+
+    $query_auth_count = sqlquery("SELECT count(*) AS count FROM $form_name " .
+                          "WHERE pid = ? AND prior_auth_number = ? ", array($pid,$query_get_authnum['prior_auth_number']));
+
+return $query_auth_count;
+}
+
+function table_exists($table)
```

Comment:
if EXISTS (select Table_Name from information_schema.TABLES where Table_Name=<TableName> and TABLE_SCHEMA=<Database Name>)
all users have access to information_schema.
LIKE comparison is very iffy.  You can also just: 
IF EXISTS(select count(id) from $table) but this will return >0 only if there is a row in the table.
Using SHOW TABLES like this is a <mysql5 method.

Reply:
ok but it still works in php 7.0

Reply:
Yeah, the issue is really just that it can cause ambiguity and is more complex than necessary.